core/editor.js

```diff
@@ -492,20 +492,21 @@
 
 	function loadPlugins( editor ) {
 		var config = editor.config,
-			plugins = config.plugins,
-			extraPlugins = config.extraPlugins,
-			removePlugins = config.removePlugins;
+			// We have to remove whitespaces (#1712).
+			plugins = config.plugins.replace( /\s/g, '' ),
+			extraPlugins = config.extraPlugins.replace( /\s/g, '' ),
```

Comment:
Now checking if such config option is set is skipped and we assume from the start that it's string. It causes several tests to fail (e.g. `tests/plugins/image2/image2`) when `extraPlugins`/`removePlugins` is set to `undefined` explicitly. Probably `replace` for every option should be moved into appropriate `if`.